### ISSUE TITLE
memory leak when using matchesJsonSchema(jsonSchema)

### ISSUE THREAD

**Issue by vguna** [rest-assured/rest-assured#736]
Hi.

I'm using RestAssured 2.9.0 for performing load tests. While running I encountered, that the test client is allocation memory that can't be released anymore. Looking at jvisualvm reveals, that using

`MatcherAssert.assertThat(jsonContent, matchesJsonSchema(jsonSchema));`

creates a load of `com.github.fge.jsonschema.core.report.ListProcessingReport` instances that won't be cleaned up by manually triggering the GC.

The test classes are using the call above, to assure that JSON responses match the given schema. The `jsonSchema` is provided as a String. Any idea what is happening?

**Comment by johanhaleby** [comment]
Thanks for reporting. See comment in #735.

**Comment by vguna** [comment]
Yes, that is reported by me :).

This is another leak that, as it looks, has nothing todo with Groovy and the referenced issue.

**Comment by johanhaleby** [comment]
Yeah I know you reported it, just wanted to link to the other issue since my comments there applies to this comment as well. Thanks again for finding the problem(s). Do you have a heap dump or anything that might help with finding the cause?
